### PR TITLE
Editing step.adoc

### PR DESCRIPTION
**Pull request by Buzzardo**
I made step.adoc more formal and easier to read for readers of varying English-language abilities. I also added a couple links and asked a couple questions in TODOs.

### CODE REVIEW

File: spring-batch-docs/asciidoc/step.adoc [hunk 1]
```diff
@@ -644,25 +648,25 @@ Transaction attributes can be used to control the isolation,
 [[registeringItemStreams]]
 
 
-==== Registering ItemStreams with the Step
+==== Registering `ItemStreams` with a `Step`
```
Comment (mminella) [inline spring-batch-docs/asciidoc/step.adoc]: The interface is `ItemStream`...

File: spring-batch-docs/asciidoc/step.adoc [hunk 2]
```diff
@@ -740,36 +744,36 @@ Any class that implements one of the extensions
 ----
 
 An `ItemReader`,
-      ItemWriter or
-      `ItemProcessor` that itself implements one of the
-      `StepListener` interfaces will be registered
+      `ItemWriter` or
+      ``ItemProcessor`` that itself implements one of the
```
Comment (mminella) [inline spring-batch-docs/asciidoc/step.adoc]: Why the double set of ticks?

File: spring-batch-docs/asciidoc/step.adoc [hunk 3]
```diff
@@ -740,36 +744,36 @@ Any class that implements one of the extensions
...
-      explicitly registered (as described above).
+      explicitly registered (as described previously under <<registeringItemStreams>>).
 
 In addition to the `StepListener` interfaces,
       annotations are provided to address the same concerns. Plain old Java
       objects can have methods with these annotations that are then converted
       into the corresponding `StepListener` type. It is
-      also common to annotate custom implementations of chunk components like
+      also common to annotate custom implementations of chunk components such as
       `ItemReader` or ItemWriter
```
Comment (mminella) [inline spring-batch-docs/asciidoc/step.adoc]: `ItemWriter` should have ticks around it.

File: spring-batch-docs/asciidoc/step.adoc [hunk 4]
```diff
@@ -944,14 +948,14 @@ public interface ItemWriteListener<S> extends StepListener {
 }
 ----
 
-The `beforeWrite` method will be called
+The `beforeWrite` method is called
         before `write` on the
-        `ItemWriter`, and is handed the item that will be
-        written. The `afterWrite` method will be called
+        `ItemWriter` and is handed the item that is
```
Comment (mminella) [inline spring-batch-docs/asciidoc/step.adoc]: "...is handed the list of items to be written."  See https://docs.spring.io/spring-batch/trunk/apidocs/org/springframework/batch/core/ItemWriteListener.html#afterWrite-java.util.List-

File: spring-batch-docs/asciidoc/step.adoc [hunk 5]
```diff
@@ -1068,17 +1072,19 @@ The `Tasklet` is a simple interface that has
 
 [NOTE]
 ====
-`TaskletStep` will automatically register the
-      tasklet as `StepListener` if it implements this
-      interface
+`TaskletStep` automatically registers the
+      tasklet as a `StepListener` if it implements the
+      interface.
+      // TODO if it implements which interface? There are a number of interfaces in this
```
Comment (mminella) [inline spring-batch-docs/asciidoc/step.adoc]: the `StepListener` interface

File: spring-batch-docs/asciidoc/step.adoc [hunk 6]
```diff
@@ -1292,56 +1298,57 @@ The "on" attribute of a transition element uses a simple
...
         flow, it is important to understand the difference between
-        BatchStatus and
-        `ExitStatus`. BatchStatus
+        `BatchStatus` and
+        `ExitStatus`. `BatchStatus`
         is an enumeration that is a property of both
-        JobExecution and
-        `StepExecution` and is used by the framework to
+        `JobExecution` and
+        ``StepExecution`` and is used by the framework to
```
Comment (mminella) [inline spring-batch-docs/asciidoc/step.adoc]: Double ticks?

File: spring-batch-docs/asciidoc/step.adoc [hunk 7]
```diff
@@ -1394,10 +1401,12 @@ public class SkipCheckingListener extends StepExecutionListenerSupport {
...
 The above code is a `StepExecutionListener`
         that first checks to make sure the `Step` was
-        successful, and next if the skip count on the
+        successful and next if the skip count on the
         `StepExecution` is higher than 0. If both
         conditions are met, a new `ExitStatus` with an
-        exit code of "COMPLETED WITH SKIPS" is returned.
+        exit code of `COMPLETED WITH SKIPS` is returned.
+// TODO "and next if the skip count on the `StepExecution` is higher than 0" confuses me.
+// Please explain that, and I'll make a stronger sentence here.
```
Comment (mminella) [inline spring-batch-docs/asciidoc/step.adoc]: Spring batch keeps track of the items that are skipped during processing.  The listener above checks if the step is successful (`if(!exitCode.equals(ExitStatus.FAILED.getExitCode())`) then it checks if the number of items skipped is more than 0 (`stepExecutino.getSkipCount() > 0`).

File: spring-batch-docs/asciidoc/step.adoc [hunk 8]
```diff
@@ -1424,59 +1428,58 @@ So far, all of the job configurations discussed have had at least
...
       `BatchStatus` or `ExitStatus`
-      of any `Steps` in the `Job`:
-      these elements will only affect the final statuses of the
+      of any `Steps` in the `Job`.
+      These elements affect only the final statuses of the
       `Job`. For example, it is possible for every step
-      in a job to have a status of FAILED but the job to have a status of
-      COMPLETED, or vise versa.
+      in a job to have a status of `FAILED` but for the job to have a status of
+      `COMPLETED`. (It is also possible for the steps to have a status of `FAILED` and the job to have a status of `COMPLETED`.)
```
Comment (mminella) [inline spring-batch-docs/asciidoc/step.adoc]: I don't understand the sentence in parens.  Isn't that echoing what the previous sentence just said?

File: spring-batch-docs/asciidoc/step.adoc [hunk 9]
```diff
@@ -644,25 +647,25 @@ Transaction attributes can be used to control the isolation,
...
 
 
-==== Registering ItemStreams with the Step
+==== Registering `ItemStream` with a `Step`
 
 The step has to take care of `ItemStream`
-      callbacks at the necessary points in its lifecycle. (for more
-      information on the `ItemStream` interface, please
-      refer to <<readersAndWriters.adoc#itemStream,itemStream>>) This is vital if a step fails,
+      callbacks at the necessary points in its lifecycle. (For more
```
[the remark this answers is not in the document]
Comment (Buzzardo) [reply spring-batch-docs/asciidoc/step.adoc]: A period should be inside the parentheses, thus: see 
<<readersAndWriters.adoc#itemStream,ItemStream>>.)

Thanks.

File: spring-batch-docs/asciidoc/step.adoc (hunk 8 above)
Comment (Buzzardo) [reply spring-batch-docs/asciidoc/step.adoc]: Yes. Remove that sentence, please. I recall that I was making sure every such place had a sentence like that, and I seem to have gotten carried away. Thanks.

**Comment by mminella** [comment]
Polished and merged as e2b18daa595870e7cc3c481722e274fc206765b0

**Pull request closed without merging**